.editorconfig: Automatically configure multiple text editors

smcv opened the pull request:
See editorconfig.org for details of the format, which is understood by
multiple text editors, either directly or via plugins.

This is not comprehensive, but should cover most of the SDL codebase.
Please extend as needed.

smcv commented:
Inspired by #5610.

This complements #5622, rather than being redundant with it. `.editorconfig` is less powerful than `.clang-format`, but also more widely available: most programmer-oriented editors have some way to read `.editorconfig`, for example vim and emacs via plugins.

madebr commented on .editorconfig:
```diff
@@ -0,0 +1,85 @@
+indent_style = space
+
+[test/Makefile.os2]
+indent_size = 2
```
```suggestion
indent_size = 4
```

madebr reviewed (commented):
To my personal projects, I also add:
```
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```
Though, I need to make sure that the trim_trailing_whitespace is not applied to autoconf generated makefiles and patches.

slouken commented:
The rules for shell scripts and `src/filesystem/psp/SDL_sysfilesystem.c` shouldn't be needed. I fixed those so they're consistent with the rest of the codebase.

smcv commented:
> trim_trailing_whitespace = true

I suspect this one is really annoying if your codebase is not already trailing-whitespace-clean, because contributors save a file after changing one line and the result is a huge diff that "fixes" all the trailing whitespace in that file, which is annoying to review and hurts the usefulness of `git blame`.

smcv commented:
> The rules for shell scripts and src/filesystem/psp/SDL_sysfilesystem.c shouldn't be needed. I fixed those so they're consistent with the rest of the codebase.

Removed those.
